src/OrleansSQLUtils/Storage/DbStoredQueries.cs

```diff
@@ -0,0 +1,406 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Data;
+using System.Linq;
+using System.Net;
+using System.Threading.Tasks;
+using Orleans;
+using Orleans.Runtime;
+using Orleans.SqlUtils;
+
+namespace OrleansSQLUtils.Storage
+{
+    internal class DbStoredQueries
+    {
+        internal static async Task<DbStoredQueries> Create(IRelationalStorage storage)
+        {
+            //this will probably be the same for all relational dbs
+            const string orleansQueriesSelectTemplate = "SELECT QueryKey, QueryText FROM OrleansQuery;";
+            var dbStoredQueries = new DbStoredQueries();
+            var propertiesDictionary = typeof (DbStoredQueries).GetFields().ToDictionary(p => p.Name, p => p);
+            var populatedKeys = await storage.ReadAsync(orleansQueriesSelectTemplate, selector =>
+            {
+                var queryKey = selector.GetValue<string>("QueryKey");
+                var queryText = selector.GetValue<string>("QueryText");
+                propertiesDictionary[queryKey].SetValue(dbStoredQueries, queryText);
+                return queryKey;
+            }, null);
+
+            //Check that all the required keys are loaded and throw an exception giving the keys expected but not loaded.
+            var missingQueryKeys = populatedKeys.Except(propertiesDictionary.Keys).ToArray();
+            if (missingQueryKeys.Length > 0)
+            {
+                throw new ArgumentException(
+                    $"Not all required queries found when loading from the database. Missing are: {string.Join(",", missingQueryKeys)}");
+            }
+            return dbStoredQueries;
+        }
+
+        private DbStoredQueries()
+        {
+        }
+
+#pragma warning disable 649
+        /// <summary>
+        /// A query template to retrieve gateway URIs.
+        /// </summary>        
+        public readonly string GatewaysQueryKey;
```

Comment:
I've a huge fan of `readonly` fields. I'm not against property accessors, since in C# 6 they act the same as `readonly` fields. So if you're more inclined toward properties, I'm OK with it. about making these `internal`, well the class is `internal` so I'm not sure if that would be beneficial. But again, I'm OK with changing to internal.